truffle-config.js

```diff
@@ -41,9 +41,11 @@ module.exports = {
     infura: {
       skipDryRun: true,
       provider: function () {
+        const infuraUrl = `${process.env.INFURA_URL}/${process.env.INFURA_API_KEY}`;
+        const cleanInfuraUrl = infuraUrl.replace(/([^:])(\/\/+)/g, '$1/');
```

Comment:
I think a better way of checking if `INFURA_URL` has a trailing slash is to use the standard `url` library instead of regexing it e.g.

```javascript
const url = require('url')

url.resolve(process.env.INFURA_URL, process.env.INFURA_API_KEY)
```

Reply:
Sounds pedantic but I'm not really a regex guru and it makes it harder to reason about 😓

Reply:
Turns out I'm completely wrong, and that is the correct way to do it!

Reply:
oh interesting, is there any document or reference on this?

Reply:
Yeah, I started off using `url` as well, but it doesn't seem to do what we want 😞 
I thought the regex was better than pulling in an external package, but feel free to do so if you feel strongly about it.

Reply:
@boolafish even the most popular url-joining library uses regex, (and its not in the std url lib) https://github.com/jfromaniello/url-join/blob/master/lib/url-join.js

So I presume its the way